rasa/model.py

```diff
@@ -340,42 +353,60 @@ def merge_model(source: Text, target: Text) -> bool:
         shutil.move(source, target)
         return True
     except Exception as e:
-        logging.debug(e)
+        logging.debug("Could not merge model: {}".format(e))
         return False
 
 
-def should_retrain(new_fingerprint: Fingerprint, old_model: Text, train_path: Text):
-    """Checks which component of a model should be retrained.
+def should_retrain(
+    new_fingerprint: Fingerprint, old_model: Text, train_path: Text
+) -> ShouldRetrain:
+    """Check which components of a model should be retrained.
 
     Args:
         new_fingerprint: The fingerprint of the new model to be trained.
         old_model: Path to the old zipped model file.
         train_path: Path to the directory in which the new model will be trained.
 
     Returns:
-        A tuple of boolean values indicating whether Rasa Core and/or Rasa NLU needs
+        A ShouldRetrain object indicating whether Rasa Core and/or Rasa NLU needs
         to be retrained or not.
 
     """
-    retrain_nlu = retrain_core = True
+    retrain = ShouldRetrain(core=True, nlu=True, nlg=True)
 
     if old_model is None or not os.path.exists(old_model):
-        return retrain_core, retrain_nlu
+        return retrain
 
     with unpack_model(old_model) as unpacked:
         last_fingerprint = fingerprint_from_path(unpacked)
-
         old_core, old_nlu = get_model_subdirectories(unpacked)
 
-        if not core_fingerprint_changed(last_fingerprint, new_fingerprint):
+        retrain = ShouldRetrain(
+            core=section_fingerprint_changed(
+                last_fingerprint, new_fingerprint, SECTION_CORE
+            ),
+            nlu=section_fingerprint_changed(
+                last_fingerprint, new_fingerprint, SECTION_NLU
+            ),
+            nlg=section_fingerprint_changed(
+                last_fingerprint, new_fingerprint, SECTION_NLG
+            ),
+        )
+
+        core_merge_failed = False
+        if not retrain.core:
             target_path = os.path.join(train_path, "core")
-            retrain_core = not merge_model(old_core, target_path)
+            core_merge_failed = not merge_model(old_core, target_path)
```

Comment:
I still don't like it that we are doing the merge here, but we can probably tackle this also in another issue

Reply:
Can we rename `merge_model` to `move_model`? Because we are not really merging anything, we are just moving the old model to the new training location.